frontend/src/router/index.ts

```diff
@@ -71,11 +79,20 @@ router.beforeEach(async (to, from) => {
   useToast().clear()
 
   // Refresh token first before navigation.
-  if (AuthService.state.value.famLoginUser) { // condition needed to prevent infinite redirect
+  if (to.meta.requiresAuth && AuthService.state.value.famLoginUser) { // condition needed to prevent infinite redirect
     await AuthService.methods.refreshToken()
   }
 })
 
+function withMeta(args: any) {
```

Comment:
Two points of confusion - this seems to add on the requiresAuth: true setting, but the name of the function (withMeta) doesn't reflect this, and this method is even used with the additional setting to set requiresAuth to false, which is weird.